hyperspy/tests/model/test_model.py

```diff
@@ -745,6 +745,31 @@ def test_one_component_multidim(self, parallel):
             s.data, np.array([np.zeros((2, 5)), np.ones((2, 5)) * 2])
         )
 
+    def test_out_of_range_to_nan(self):
+        index = 2
+        self.m.channel_switches[:index] = False
+        s1 = self.m.as_signal(component_list=[0], out_of_range_to_nan=True,
+                              parallel=False)
+
+        s2 = self.m.as_signal(component_list=[0], out_of_range_to_nan=False,
```

Comment:
Indeed, as mentioned by @tjof2 in #2507, threading is playing up here, if I use `parallel=True`, the `channel_switches` are changed in an unexpected manner.

Reply:
Another argument for redoing it. #2503 is only really a temporary fix by removing it (deprecating it really) such that it can be returned to.

Reply:
If this passes tests, let's merge, then I'll fix the conflicts for #2503 and continue the discussion there with regards to keeping/changing/removing the parallel argument.

Reply:
> Another argument for redoing the parallel code. #2503 is only really a temporary fix by removing it (deprecating it really) such that it can be returned to. It's really not a great implementation of parallelism at the moment.

Agreed!